### DISCUSSION TITLE
Game/Unity crashes randomly and closes my game

### DISCUSSION THREAD

**Discussion by DaDonut** [voxeltycoon/issues#940] [Troubleshooting] [answered]
**Describe the bug**
Randomly when I'm playing the game, a Unity Voxel Tycoon crash screen will pop up and crash my game. This happens for no reason as I can tell, sometimes I'll just be idly watching the flow of traffic and it will crash, and other times I'll be laying down roads.

I have updated my graphics drivers and uninstalled the game with no fix.
![4ec9c9014bfb978f3b8472980606ec43](https://user-images.githubusercontent.com/57339251/119204302-b9164700-ba5a-11eb-954e-bd9ed84a4225.png)

**Steps to reproduce**
I'm not sure, just play the game

**Build version**
0.86.0.0 Steam

**Platform**
Windows 10

**Reproducibility**
It happens randomly for no reason, sometimes I can play for hours and nothing happens. Other times it crashes 4 times in 30 minutes.

**Logs/saves**
[Player.log](https://github.com/voxeltycoon/issues/files/6525323/Player.log)

**Comment by CasianoKoprowski** [comment]
Hello! I thought I'd plus one this issue, but I'll add that it's pretty similar to #764 as well.  Unlike the OP in this ticket and more like #764, I rarely get more than 10-20 minutes of playtime per session.  Sometimes it crashes immediately after loading into a game.  I do have Steam Workshop mods enabled, but this issue was present before I installed any mods.  

-Edit- Added another Player-prev.log and my entire crashes folder

**Steps to reproduce**
Play the game.  Sometimes it seems like it's related to placing infrastructure (buildings, roads/tracks) and sometimes it just happens.  

**Build version**
0.86.0.0 Steam

**Platform**
Windows 10

**Reproducibility**
Playing the game for any amount of time will result in the crash

**Logs/saves**
[20210608_Player-prev.log](https://github.com/voxeltycoon/issues/files/6641676/20210608_Player-prev.log)
[20210611_Player-prev.log](https://github.com/voxeltycoon/issues/files/6641692/20210611_Player-prev.log)
[Crashes.zip](https://github.com/voxeltycoon/issues/files/6641693/Crashes.zip)

**Comment by MrKiel** [comment]
~~Same!~~
EDIT: Unity/XMP Issue https://github.com/voxeltycoon/issues/issues/796#issuecomment-873870963

Appears to me a Unity access violation. 
Putting the crash.dmp through a debugger shows no useful information, unfortunately (that I can see, anyway). 
[crashdumpanalysis.txt](https://github.com/voxeltycoon/issues/files/6736796/crashdumpanalysis.txt)


[crash.zip](https://github.com/voxeltycoon/issues/files/6736781/crash.zip)
[error.log](https://github.com/voxeltycoon/issues/files/6736779/error.log)
[Player.log](https://github.com/voxeltycoon/issues/files/6736780/Player.log)

**Comment by CasianoKoprowski** [comment]
Just wanted to add that this continues for me in 0.86.1 after I noticed that patch note that read: "Unity updated to 2020.3.12"

**Comment by MrKiel** [comment]
Withdrawing my +1. 

In my case, Unity is crashing under oddly specific conditions present in XMP-overclocked RAM at 3200Mhz. To be clear: there's nothing wrong with the RAM, the issue lies with either the XMP profile being 'too loose' or Unity introducing unique failure conditions that other applications / game engines do not. 

Source used for troubleshooting [here](https://forum.unity.com/threads/unityplayer-dll-every-games-made-with-it-are-randomly-crashing.897173/)

My solution [here](https://forum.kerbalspaceprogram.com/index.php?/topic/203570-unity-crashing-access-violation/)

To summarise: It took me months to track this down, as I only recently realised that it was only Unity games, including this one, that were crashing with this error. I'm still not 100% sure it's a fix, given that I've had the problem for months and the crashes are not reproducible (seemingly random). However, I've had zero crashes in Unity games since my attempted fix, having played >10 hours of Voxel Tycoon and >10 hours of Kerbal without issues, which is far better than it was. 

I manually entered the RAM timings in the BIOS for 3200Mhz using Ryzen DRAM Calculator (see solution above), but choosing another XMP profile (not 3200Mhz) may also resolve the issue. 

Note that this affects both AMD and Intel CPU (see source thread linked above to Unity forums), Intel CPUs have a similar solution.

**Comment by MrKiel** [comment]
@CasianoKoprowski and @DaDonut hopefully the above will help you

**Comment by CasianoKoprowski** [comment]
Oh; interesting!  I'll have to set aside some time to read through everything and try this out.  I do in fact have an XMP profile active.

Thanks for the find @MrKiel

**Comment by andrewpey** [comment]
Did the solution work for you guys?

**Comment by DaDonut** [comment]
> Did the solution work for you guys?

I tried MrKiel’s solution but it didn’t seem to work, the same crash screen still happened. Odd thing is, I didn’t have my RAM over locked or on XMP mode

**Comment by andrewpey** [comment]
Sorry to hear this, still nothing points out the problem with the game itself, more like a system/drivers/hardware issue. So, there's not much we can do with it, unfortunately.

Moving this to discussions, for an ongoing conversation in case there will be some new info or ideas.

**Comment by MrKiel** [comment, marked as answer]
Howdy,

Apologies for resurrecting this old thread.
Bottom line: it was a defective CPU causing intermittent crashes in some unity games under certain conditions, exacerbated by overclocked Ram, for example.

I recently replaced my cpu and that has resolved all unity-based crashes. 

Some late-2018 Ryzen CPUs have a very niche timing issue that only seems to show up in some pretty unique scenarios. Not all CPUs from that era were affected, but googling "ryzen unity crash" will give you a rough idea. 

This drove me nuts for a full two years before I finally bit the bullet and decided to replace the CPU, and everything (including this game) works fine now. I can only apologise for incorrectly thinking that only some specific unity games were at fault, rather than my hardware.

**Comment by andrewpey** [comment]
Thanks for getting back to the thread and letting us know! We are glad that the issue is finally resolved.
